library/src/test/java/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveDataTests.kt

```diff
@@ -0,0 +1,122 @@
+package com.getstream.sdk.chat.viewmodel.messages
+
+import androidx.arch.core.executor.testing.InstantExecutorExtension
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.getstream.sdk.chat.adapter.MessageListItem
+import com.getstream.sdk.chat.createMessage
+import com.getstream.sdk.chat.view.messages.MessageListItemWrapper
+import io.getstream.chat.android.client.models.ChannelUserRead
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldNotBeEqualTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(InstantExecutorExtension::class)
+class MessageListItemLiveDataTests {
+
+    private val currentUser = User()
+    private lateinit var messages: MutableLiveData<List<Message>>
+    private lateinit var threadMessages: MutableLiveData<List<Message>>
+    private lateinit var typing: MutableLiveData<List<User>>
+    private lateinit var reads: MutableLiveData<List<ChannelUserRead>>
+
+    private lateinit var observer: TestObserver<MessageListItemWrapper>
+
+    private lateinit var sut: MessageListItemLiveData
+
+    @BeforeEach
+    fun setUp() {
+        observer = TestObserver()
+        messages = MutableLiveData(listOf())
+        threadMessages = MutableLiveData(listOf())
+        typing = MutableLiveData(listOf())
+        reads = MutableLiveData(listOf())
+
+        sut = MessageListItemLiveData(currentUser, messages, threadMessages, typing, reads)
+    }
+
+    @Test
+    fun `When messages are changed if live data in the thread state should not be observed any values `() {
+        sut.observeForever(observer)
+        threadMessages.value = listOf(createMessage())
+        observer.cleanState()
+
+        messages.value = listOf(createMessage())
+
+        observer.lastObservedValue shouldBeEqualTo null
+    }
+
+    @Test
+    fun `When messages are changed if live data is not in the thread state should observe last assigned value to messages`() {
+        sut.observeForever(observer)
+        val sendingMessage = createMessage()
+
+        messages.value = listOf(sendingMessage)
+
+        observer.lastObservedValue.also { result ->
```

Comment:
In a context like this, when you want to introduce a name for a value, `let` is usually the function used instead of `also`, as a convention in Kotlin.

In this specific case, just going with `val result = observer.lastObservedValue` seems to be good enough too, and it would simplify the code.

Reply:
As I see `also` is to apply some side-effect to our computation.
On that case, it is not a side-effect, it is part of our check-task.
I will go with 
```
val result = observer.lastObservedValue
result.shouldNotBeNull()
....
```

Reply:
Thanks guys, fixed it